### PR TITLE
feat(sdui-runtime): add usePageStore.replaceNode + appendItems

### PR DESCRIPTION
**Pull request by mridulgupta-oi**
## Summary

- Add `page: Page | null` state to `usePageStore` (the full SDUI page envelope).
- Add `setPageTree(page)` setter for the initial page fetch.
- Add `replaceNode(targetId, next)` — recursive node-tree replacement by id.
- Add `appendItems(targetId, items, options?)` — append to `data.items` and merge optional `cursor` / `has_more` into the target's `data`.
- Extract pure tree-walkers (`replaceNodeInTree`, `appendItemsInTree`) into an `__internal` namespace and add 13 unit tests covering happy paths, no-op fallbacks, and warning emission.

## Why

Three action handlers were calling methods on `usePageStore` that did not exist, silently crashing the action dispatch loop:

| Handler | Call site | Method called |
|---|---|---|
| `handleReloadSection` | `src/action-engine/handlers/reload-section.ts:40` | `replaceNode(target, node)` |
| `handleReplaceSection` | `src/action-engine/handlers/replace-section.ts:16` | `replaceNode(target, with_node)` |
| `handleAppendItems` | `src/action-engine/handlers/append-items.ts:17` | `appendItems(target, items, { cursor, hasMore })` |

The store previously only exposed section-granularity setters (`replaceSection`, `appendSection`) keyed by a flat sections map. The action handlers, however, address nodes by id anywhere in the page tree — which is what these new methods now support. Both methods are no-ops with `console.warn` when the target id is missing or has no `data.items` array, so stale actions degrade gracefully.

## Consumer files

- `packages/sdui-runtime/src/action-engine/handlers/reload-section.ts`
- `packages/sdui-runtime/src/action-engine/handlers/replace-section.ts`
- `packages/sdui-runtime/src/action-engine/handlers/append-items.ts`

## Test plan

- [x] `npm test -w packages/sdui-runtime` — 13 new tests pass, no existing tests regressed (3 pre-existing baseline failures in `branch.test.ts` / `compound.test.ts` / `set-local.test.ts` are unchanged and unrelated to this PR).
- [x] `npm run build -w packages/sdui-runtime` succeeds with no type errors.
- [ ] Manual verification: trigger `reload_section`, `replace_section`, and `append_items` actions from the creator-app simulator and confirm the page tree updates without dispatch errors.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/sdui-runtime/src/state/usePageStore.ts [hunk 1]
```diff
@@ -106,6 +253,57 @@ export const usePageStore = create<PageStoreState & PageStoreActions>((set) => (
       };
     }),
 
+  replaceNode: (targetId, next) =>
+    set((state) => {
+      if (!state.page) {
+        console.warn(
```
Comment (zenith-oi) [inline packages/sdui-runtime/src/state/usePageStore.ts:259]: **FLAG** — [pii-logging-discipline / no-console-in-production] The store uses `console.warn(...)` directly in four places (lines 259–261, 270–272, 281–283, 293–295, 299–301). The Zenith coding standards say 'Never use `print()` or `console.log` in production' and the pii-logging-discipline pattern extends this to all console calls. For a shared runtime package consumed by three products, bare `console.warn` is hard to silence in production builds and can't be routed through a structured logger or suppressed by the host app. Consider accepting an optional `logger` dependency (defaulting to a no-op or `console`) via the store's initialisation, or at minimum gating the warns behind a `__DEV__` / `process.env.NODE_ENV !== 'production'` check so production bundles stay silent.

**Review by zenith-oi — DISMISSED** [review]
## Zenith Code Review

**Clean, focused store extension that unblocks three action handlers. The implementation is correct, the tests are thorough, and the no-op/warn degradation path is well-handled. One minor concern worth noting: `console.warn` is used directly in production store code rather than a structured logger, which is a mild violation of the coding standards' 'never use console.log in production' rule — but given this is a frontend runtime package (not a backend service), the severity is low.**

### What this PR is doing
**Intent:** Add `page: Page | null` state, `setPageTree`, `replaceNode`, and `appendItems` actions to `usePageStore`, plus 13 unit tests covering the new tree-walker logic — fixing three action handlers that were silently crashing because they called store methods that didn't exist.
**Delivered:** Fully delivered. The diff adds all four new store members described in the PR body, extracts the two pure tree-walker functions into an `__internal` export for testability, and ships 13 tests covering happy paths, no-op fallbacks, and warning emission. The test script in `package.json` is updated to include the new test file. No linked task page or brief is present, so there are no numbered acceptance items to check against.

**1 FLAG** — advisory, not merge-blocking. See inline comments.

**Positive:**
- The `__internal` export pattern for unit-testing pure functions without polluting the public API is exactly right — it keeps the public surface clean while making the tree-walkers fully testable.
- The three-tuple return `[nodes, matched, mutated]` from `appendItemsInTree` cleanly separates 'found the node' from 'actually mutated it', enabling precise warning messages at the store layer.
- Immutable update discipline is solid throughout — unmodified siblings are preserved by reference (verified in tests at lines 41–42), which means React/Zustand shallow-equality checks will correctly skip re-renders for unchanged subtrees.
- Test coverage is genuinely comprehensive: happy paths, deep nesting, no-op when id missing, no-op when target has no `data.items`, and warning emission are all exercised with reference-equality assertions where appropriate.

---
*Generated by [Zenith AI CTO](https://github.com/one-impression/zenith-coding-agent)*

**Review by zenith-oi — DISMISSED** [review]
## Zenith Code Review

**Clean, well-scoped PR that fills a real gap (three action handlers calling methods that didn't exist). The implementation is correct, immutable, and well-tested. No issues worth blocking or flagging.**

### What this PR is doing
**Intent:** Add `page: Page | null` state, `setPageTree`, `replaceNode`, and `appendItems` actions to `usePageStore` so that the three SDUI action handlers (`handleReloadSection`, `handleReplaceSection`, `handleAppendItems`) that were already calling these non-existent methods no longer silently crash, and cover the new logic with 13 unit tests (inferred from diff + body).
**Delivered:** Fully delivered. The diff adds all four described API surface items (`page` field, `setPageTree`, `replaceNode`, `appendItems`), extracts the two pure tree-walkers into an `__internal` namespace, wires the test file into the `package.json` test script, and ships 13 tests covering happy paths, no-op fallbacks, and warning emission. No linked task page or brief was provided, so there are no numbered acceptance items to check against.

No issues found.

**Positive:**
- Immutable tree-walk pattern is correct: unmodified siblings are preserved by reference, which means React/Zustand shallow-equality checks will correctly skip re-renders for unchanged subtrees.
- The `__internal` export pattern (clearly doc-commented as test-only) is a clean way to unit-test pure functions without polluting the public API surface.
- Warning messages include the node id and page id, making debugging stale action dispatches straightforward.
- The `matched` / `mutated` two-flag return from `appendItemsInTree` correctly distinguishes 'target not found' from 'target found but has no data.items', enabling distinct warning messages at the store layer — a subtle but important UX distinction for debugging.
- Test coverage is thorough: reference-equality assertions on unmodified siblings verify the immutability contract, not just value equality.

---
*Generated by [Zenith AI CTO](https://github.com/one-impression/zenith-coding-agent)*

File: packages/sdui-runtime/src/state/usePageStore.ts (hunk 1 above)
Comment (zenith-oi) [inline packages/sdui-runtime/src/state/usePageStore.ts:259]: **FLAG** — [pii-logging-discipline / no-console-in-production] The store uses `console.warn(...)` directly in four places (lines 259–261, 270–272, 281–283, 293–295, 299–301). The Zenith coding standard says 'Use `logging` (Python) or `logger` from config (TypeScript). Never use `print()` or `console.log` in production.' `console.warn` is in the same family. In a React Native / bundled context these calls also survive into production builds with no filtering or structured context. The fix is to accept an optional `logger` dependency (or use a no-op logger by default) so callers can wire in their structured logger and the store doesn't reach for the global console. This was flagged in the previous round and remains unaddressed.

**Review by zenith-oi — DISMISSED** [review]
## Zenith Code Review

**Clean, well-scoped addition of node-tree mutation methods to `usePageStore`. The implementation is correct, immutable, and well-tested. One carry-over FLAG from the previous round remains unaddressed: `console.warn` is used directly in four store action paths instead of a structured logger, which violates the Zenith coding standard against `console.*` in production code.**

### What this PR is doing
**Intent:** Add `page: Page | null` state, `setPageTree()`, `replaceNode()`, and `appendItems()` methods to `usePageStore` so that the three action handlers (`handleReloadSection`, `handleReplaceSection`, `handleAppendItems`) that were silently crashing due to calling non-existent store methods now have real implementations backed by immutable recursive tree-walkers.
**Delivered:** Fully delivered. The diff adds all four declared additions (`page` field, `setPageTree`, `replaceNode`, `appendItems`), extracts the two pure tree-walkers into an `__internal` namespace, and ships 13 unit tests covering happy paths, no-op fallbacks, reference-preservation, and warning emission. The changeset file is present and correctly scoped as `minor`. No scope drift detected — the PR does exactly what the body describes and nothing more.

> ℹ️ 1 previous Zenith review dismissed. Inline comments from those reviews are superseded — only comments from **this review** are current.

**1 FLAG** — advisory, not merge-blocking. See inline comments.

**Positive:**
- Immutable tree-walk implementation is correct: unmodified siblings preserve object identity (`out.push(n)` on the non-matching branch), which is exactly what React's shallow-equality bail-out needs — and the tests explicitly assert reference equality on siblings.
- The `__internal` export pattern for unit-testing pure functions without polluting the public API surface is a clean idiom — keeps the public contract tight while making the walkers fully testable.
- Test coverage is thorough: happy path, deep nesting, no-op when id missing, no-op when `data.items` is absent, cursor/has_more merge, warning emission, and no-page-loaded guard — all 13 cases are meaningful and non-redundant.
- Backward compatibility is preserved: the existing `sections`/`replaceSection`/`appendSection` API is untouched, and `setPageTree` sets `pageId` from `page.id` so both layers stay in sync.
- The changeset description is unusually informative — it explains the root cause (silent crashes), the fix (immutable tree walks), and the React perf implication (reference preservation) in one paragraph.

---
*Generated by [Zenith AI CTO](https://github.com/one-impression/zenith-coding-agent)*

**Review by zenith-oi — DISMISSED** [review]
## Zenith Code Review

**Clean, well-scoped fix that closes a silent crash loop in the SDUI action dispatch pipeline. The implementation is correct, the tree-walkers are properly immutable (unmodified siblings preserve reference identity for React's shallow-equality bail-out), and the test coverage is thorough. No issues worth blocking or flagging.**

### What this PR is doing
**Intent:** Add `page: Page | null` state, `setPageTree()`, `replaceNode()`, and `appendItems()` methods to `usePageStore` so that the three action handlers (`handleReloadSection`, `handleReplaceSection`, `handleAppendItems`) that were already calling these non-existent methods stop silently crashing on every dispatch (inferred from diff + body).
**Delivered:** Fully delivered. The diff adds all four described API surfaces, implements immutable recursive tree-walkers for both replace and append operations, exports them via `__internal` for unit testing, and ships 13 tests covering happy paths, no-op fallbacks, reference-preservation, and warning emission. The changeset entry is present and correctly classified as `minor`. No scope overshoot detected — the legacy `sections`/`replaceSection`/`appendSection` API is untouched.

No issues found.

**Positive:**
- Reference-identity preservation for unmodified siblings is explicitly tested (`assert.equal(next[0], tree[0])`) — this is the right thing to do for React performance and the author verified it rather than assuming it.
- The `__internal` export pattern (test-only escape hatch, clearly documented) lets the pure tree-walkers be unit-tested without making them part of the public API — good discipline.
- Both action methods degrade gracefully with `console.warn` rather than throwing, which is the correct behaviour for stale/mismatched action payloads in a UI runtime.
- The `appendItemsInTree` three-tuple return `[nodes, matched, mutated]` correctly distinguishes 'target not found' from 'target found but has no data.items array', and both cases are tested and warned on separately — avoids a subtle silent no-op.

---
*Generated by [Zenith AI CTO](https://github.com/one-impression/zenith-coding-agent)*

File: packages/sdui-runtime/src/state/usePageStore.ts (hunk 1 above)
Comment (zenith-oi) [inline packages/sdui-runtime/src/state/usePageStore.ts:259]: **FLAG** — [pii-logging-discipline / no-console-in-production] The store uses `console.warn(...)` directly in five places (lines 259–261, 270–272, 281–283, 293–295, 299–301). The Zenith coding standard says 'Use `logger` from config (TypeScript). Never use `print()` or `console.log` in production.' `console.warn` is the same class of violation — it bypasses structured logging, can't be silenced in test environments without monkey-patching (as the tests themselves demonstrate with `console.warn = (...) => ...`), and leaks to stdout in production with no correlation IDs. The fix is to inject or import the repo's structured logger and call `logger.warn(...)` instead. If no logger is available in this package, at minimum wrap behind a `__DEV__` / `process.env.NODE_ENV !== 'production'` guard so production bundles are silent.

**Review by zenith-oi — DISMISSED** [review]
## Zenith Code Review

> ⚠️ **Round 3 — engineer attention requested.** The reviewer-engineer loop has reached round 3; this is the threshold where auto-iteration typically stops being productive (either the review is missing context the engineer has, or there's a genuine design call to make). Auto-fixes are still posted, but please review this thread end-to-end rather than relying on another round to converge.

**This PR correctly implements the missing `replaceNode` and `appendItems` store methods that were causing silent crashes in three action handlers. The implementation is clean — immutable tree walks, reference-preserving unmodified siblings, graceful no-ops with warnings, and solid test coverage. One carry-over issue from the previous review round remains unresolved: direct `console.warn` calls in production store code.**

### What this PR is doing
**Intent:** Add `page: Page | null` state, `setPageTree()`, `replaceNode()`, and `appendItems()` methods to `usePageStore` so that the three action handlers (`handleReloadSection`, `handleReplaceSection`, `handleAppendItems`) that were calling non-existent store methods no longer silently crash on dispatch (inferred from diff + body).
**Delivered:** Fully delivered. The four new store members are implemented with correct immutable tree-walk semantics, reference-preserving no-op paths, and `console.warn` degradation on miss. The pure tree-walker functions are extracted and exported via `__internal` for unit testing. 13 tests cover happy paths, no-op fallbacks, and warning emission. The test script in `package.json` is updated to include the new test file. The changeset file is present and correctly scoped as a minor bump. No acceptance items from a linked task page to check against.

> ℹ️ 1 previous Zenith review dismissed. Inline comments from those reviews are superseded — only comments from **this review** are current.

**1 FLAG** — advisory, not merge-blocking. See inline comments.

**Positive:**
- Immutable tree-walk design is correct: unmodified siblings are pushed by reference, so React's shallow-equality bail-out works as intended — this is explicitly tested and the changeset calls it out.
- The `[newNodes, matched, mutated]` triple-return from `appendItemsInTree` cleanly separates 'found the node' from 'actually mutated it', enabling the two distinct warning paths in the store action without re-walking the tree.
- Exporting pure functions via `__internal` rather than making them module-level exports keeps the public API clean while still enabling direct unit testing of the tree-walk logic — a good pattern for this kind of utility.
- Test coverage is thorough: every no-op branch (id not found, target has no `data.items`, no page loaded) has a dedicated test that asserts both the reference-equality invariant and the warning message content.

---
*Generated by [Zenith AI CTO](https://github.com/one-impression/zenith-coding-agent)*

File: packages/sdui-runtime/src/state/usePageStore.ts (hunk 1 above)
Comment (zenith-oi) [inline packages/sdui-runtime/src/state/usePageStore.ts:259]: **FLAG** — [pii-logging-discipline / no-console-in-production] The Zenith coding standards say 'Never use `console.log` in production' and to use a configured `logger` instead. The four `console.warn` calls added here (lines 259, 270, 281, 293, 299) are in a library package that ships to production apps. In a React Native / mobile context these warnings will surface in device logs and may be stripped by bundlers inconsistently. The fix is low-friction: accept an optional `logger` parameter on the store creator, or export a `setLogger` hook so consumers can route warnings through their own structured logger. If the team has a deliberate policy of using `console.warn` for degraded-gracefully paths in this package, a one-line comment explaining that decision would satisfy the concern.

**Review by zenith-oi — DISMISSED** [review]
## Zenith Code Review

**Solid, focused fix for a silent crash class: three action handlers were calling store methods that didn't exist. The implementation is immutable, reference-preserving, and well-tested. No blocking issues found; one minor observation about `console.warn` in production code noted below.**

### What this PR is doing
**Intent:** Add `page: Page | null` state, `setPageTree()`, `replaceNode()`, and `appendItems()` methods to `usePageStore` so that the three action handlers (`handleReloadSection`, `handleReplaceSection`, `handleAppendItems`) that were already calling these non-existent methods stop silently crashing, plus add 13 unit tests covering the new tree-walker logic and store integration.
**Delivered:** Fully delivered. The diff adds all four declared items (`page` field, `setPageTree`, `replaceNode`, `appendItems`), exports the pure tree-walkers via `__internal` for testing, wires the test file into the `package.json` test script, and ships 13 tests covering happy paths, no-op fallbacks, reference-preservation, and warning emission. The changeset file is present and correctly classified as `minor`. No scope overshoot detected.

**1 FLAG** — advisory, not merge-blocking. See inline comments.

**Positive:**
- Immutable tree-walk with reference preservation for unmodified siblings is exactly the right approach for React's shallow-equality bail-out — the changeset description even calls this out explicitly, showing the author understood the performance implication.
- The `[newNodes, matched, mutated]` triple-return from `appendItemsInTree` cleanly separates 'found the target' from 'actually changed something', enabling precise warning messages without re-walking the tree.
- Exporting pure functions via `__internal` rather than making them module-level exports keeps the public API clean while still enabling direct unit testing of the tree-walkers — good pattern for a library package.
- Test coverage is thorough: reference-equality assertions on unmodified siblings, no-op detection, warning-text assertions, and the edge case of a container node with no `data.items` array are all covered.

**Read against:**
- `PR body — consumer files table (reload-section.ts:40, replace-section.ts:16, append-items.ts:17)`
- `PR body — test plan (13 tests, build succeeds, manual verification deferred)`

---
*Generated by [Zenith AI CTO](https://github.com/one-impression/zenith-coding-agent)*

File: packages/sdui-runtime/src/state/usePageStore.ts (hunk 1 above)
Comment (zenith-oi) [inline packages/sdui-runtime/src/state/usePageStore.ts:259]: **FLAG** — [pii-logging-discipline] Five `console.warn(...)` calls remain in production store code (lines 259–261, 270–272, 281–283, 293–295, 299–301). The Zenith coding standard says 'Use `logger` from config (TypeScript). Never use `print()` or `console.log` in production.' `console.warn` is in the same category. In a React Native / bundled context these calls also survive minification and can surface in production bundles. The fix is to accept an optional `logger` parameter on the store creator (or import a shared no-op logger) and route warnings through it, which also makes the warning behaviour testable without monkey-patching `console.warn` as the tests currently do.

Comment (zenith-oi) [inline packages/sdui-runtime/src/state/usePageStore.ts:259]: **FLAG** — [pii-logging-discipline / no-console-in-production] Five `console.warn(...)` calls remain in production store code (lines 259–261, 270–272, 281–283, 293–295, 299–301). The Amplify coding standard says 'Use `logging` (Python) or `logger` from config (TypeScript). Never use `print()` or `console.log` in production.' The same principle applies to `console.warn` — it bypasses any structured logging pipeline, can't be silenced in test environments without monkey-patching (as the tests themselves demonstrate), and leaks to end-user consoles in React Native bundles. Replace with a lightweight logger abstraction (e.g. a `logger` imported from a shared util, or at minimum a module-level `const logger = { warn: (...args) => { if (process.env.NODE_ENV !== 'production') console.warn(...args); } }` guard). This was flagged in the previous round and is still unaddressed.

**Review by zenith-oi — APPROVED** [review]
## Zenith Code Review

> ⚠️ **Round 4 — engineer attention requested.** The reviewer-engineer loop has reached round 4; this is the threshold where auto-iteration typically stops being productive (either the review is missing context the engineer has, or there's a genuine design call to make). Auto-fixes are still posted, but please review this thread end-to-end rather than relying on another round to converge.

**Solid, focused fix for a silent crash class in the SDUI action dispatch loop. The tree-walker logic is correct, the immutability contract is well-tested, and the existing `sections`/`replaceSection`/`appendSection` API is untouched. One carry-over FLAG from the previous round remains unaddressed: `console.warn` is used directly in production store code rather than a structured logger.**

### What this PR is doing
**Intent:** Add `page: Page | null` state, `setPageTree()`, `replaceNode()`, and `appendItems()` methods to `usePageStore` so that the three action handlers (`handleReloadSection`, `handleReplaceSection`, `handleAppendItems`) that were silently crashing due to calling non-existent store methods now have real implementations backed by immutable recursive tree-walkers.
**Delivered:** Fully delivered. The diff adds all four advertised API surfaces, implements the two pure tree-walker functions with correct immutable-update semantics (unmodified siblings preserve reference identity), wires them into the Zustand store with appropriate no-op + warn guards, exports them via `__internal` for testing, adds 13 unit tests covering happy paths and all no-op branches, and registers the new test glob in `package.json`. The changeset file is present and correctly classified as `minor`. No scope overshoot detected.

> ℹ️ 1 previous Zenith review dismissed. Inline comments from those reviews are superseded — only comments from **this review** are current.

**1 FLAG** — advisory, not merge-blocking. See inline comments.

**Positive:**
- Immutable tree-walk with reference-preserving sibling pass-through is exactly the right approach for React shallow-equality bail-out — the changeset description calls this out explicitly and the tests verify it with `assert.equal(next[0], tree[0])`.
- The `[newNodes, matched, mutated]` triple-return from `appendItemsInTree` cleanly separates 'found the node' from 'actually mutated it', enabling the two distinct warning paths in the store action without re-walking the tree.
- Exporting tree-walkers via `__internal` rather than making them top-level exports is the right encapsulation choice — keeps the public API surface clean while still enabling direct unit testing of the pure functions.
- Test coverage is thorough: every no-op branch (id not found, target has no `data.items`, no page loaded) has its own test, and the reference-identity assertions on unmodified siblings are a meaningful correctness check, not just coverage padding.

**Read against:**
- `PR body — consumer call sites listed (reload-section.ts:40, replace-section.ts:16, append-items.ts:17)`
- `PR body — test plan (13 tests, build passes, 3 pre-existing baseline failures unrelated)`

---
*Generated by [Zenith AI CTO](https://github.com/one-impression/zenith-coding-agent)*

**Review by zenith-oi — APPROVED** [review]
## Zenith Code Review

> ⚠️ **Round 4 — engineer attention requested.** The reviewer-engineer loop has reached round 4; this is the threshold where auto-iteration typically stops being productive (either the review is missing context the engineer has, or there's a genuine design call to make). Auto-fixes are still posted, but please review this thread end-to-end rather than relying on another round to converge.

**Clean, well-scoped PR that fills a real gap (silent crashes in three action handlers). The implementation is correct, the immutable tree-walk is sound, and the test coverage is thorough. One carry-over FLAG from the previous round remains unaddressed: `console.warn` is used directly in five places in production store code, which violates the Amplify coding standard against `console.*` in production.**

### What this PR is doing
**Intent:** Add `page: Page | null` state, `setPageTree()`, `replaceNode()`, and `appendItems()` methods to `usePageStore` so that the three action handlers (`handleReloadSection`, `handleReplaceSection`, `handleAppendItems`) that were silently crashing due to calling non-existent store methods now have real implementations backed by immutable recursive tree-walkers.
**Delivered:** Fully delivered. The diff adds all four new API surface items described in the PR body, extracts the pure tree-walker functions into an `__internal` namespace for testability, adds 13 unit tests covering happy paths, no-op fallbacks, and warning emission, wires the new test glob into `package.json`, and ships a changeset entry. The existing `sections`/`replaceSection`/`appendSection` API is untouched. No acceptance items from a linked task page are present, but the PR body's stated goals are all addressed.

**1 FLAG** — advisory, not merge-blocking. See inline comments.

**Positive:**
- Immutable tree-walk design is correct and well-reasoned: unmodified siblings preserve object identity, which is exactly what React's shallow-equality bail-out needs. The `matched` early-exit optimization avoids unnecessary allocations after the target is found.
- The `__internal` export pattern for unit-testing pure functions without polluting the public API is a clean approach — the JSDoc warning discourages misuse from app code.
- Test coverage is genuinely thorough: happy paths, no-op when id missing, no-op when target has no `data.items`, warning emission for every failure mode, and reference-equality assertions for the immutability guarantee. The store-action integration tests also verify the no-op-returns-same-reference invariant, which is the React performance contract.
- Backward compatibility is preserved — the existing `sections`/`replaceSection`/`appendSection` surface is untouched, and `setPageTree` also sets `pageId` so both layers stay in sync.
- Changeset entry is accurate and descriptive, correctly classifying this as a `minor` bump.

---
*Generated by [Zenith AI CTO](https://github.com/one-impression/zenith-coding-agent)*

**Review by zenith-oi — APPROVED** [review]
## Zenith Code Review

**This PR correctly implements the missing `replaceNode` and `appendItems` methods on `usePageStore`, fixing a silent crash in three action handlers. The implementation is clean, the tree-walkers are immutable and reference-preserving, and the test coverage is thorough. No blocking issues found.**

### What this PR is doing
**Intent:** Add `page: Page | null` state, `setPageTree()`, `replaceNode()`, and `appendItems()` methods to `usePageStore` so that the three action handlers (`handleReloadSection`, `handleReplaceSection`, `handleAppendItems`) that were already calling these methods have a real implementation to dispatch to instead of silently crashing.
**Delivered:** Fully delivered. The diff adds all four new API surface items described in the PR body, implements immutable tree-walk helpers (`replaceNodeInTree`, `appendItemsInTree`) with correct reference-preservation for unmodified siblings, wires them into the Zustand store with appropriate no-op + warn guards, exports them via `__internal` for unit testing, and ships 13 tests covering happy paths, no-op fallbacks, and warning emission. The changeset file is present and correctly classified as `minor`. The test script in `package.json` is updated to include the new test file. No acceptance items from a linked task page are present, but the PR body's stated deliverables are all present in the diff.

No issues found.

**Positive:**
- Immutable tree-walk with reference preservation for unmodified siblings is exactly right for React's shallow-equality bail-out — the changeset description even calls this out explicitly, showing the author understood the React rendering implication.
- The `__internal` export pattern (clearly documented as test-only, not part of the public API) is a clean way to unit-test pure helpers without polluting the module's public surface.
- The `matched` / `mutated` distinction in `appendItemsInTree` correctly separates 'target not found' from 'target found but has no data.items array', enabling two distinct warning messages that will be genuinely useful for debugging stale actions.
- Backward compatibility is preserved — the existing `sections`/`replaceSection`/`appendSection` API is untouched, and `setPageTree` sets `pageId` as a side-effect so both layers stay in sync.

**Read against:**
- `PR body — table of three handler call sites that were silently crashing`
- `PR body — stated deliverables: page field, setPageTree, replaceNode, appendItems, __internal, 13 tests`
- `PR body — test plan checklist`

---
*Generated by [Zenith AI CTO](https://github.com/one-impression/zenith-coding-agent)*
